lib/pipeline.js

```diff
@@ -799,11 +800,33 @@ class PipelineModel extends BaseModel {
         /* eslint-enable global-require */
         const factory = SecretFactory.getInstance();
 
-        const secrets = factory.list(listConfig);
+        let secrets = factory.list(listConfig);
+
+        // Fetch and merge config pipeline's secrets
+        if (this.configPipelineId && this.configPipelineId !== this.id) {
+            const configPipelineListConfig = hoek.clone(listConfig);
+
+            configPipelineListConfig.params.pipelineId = this.configPipelineId;
+
+            return Promise.all([
+                secrets,
```

Comment:
can probably delete line 803 and just put ~secrets~`factory.list(listConfig),` here

Reply:
Sadly, in the case of regular pipelines we will still need to return `secrets` on line `835`

Reply:
Ah I missed that. Nvm!